src/aoptk/abbreviation_translator_dictionary.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+import re
+from aoptk.abbreviation_translator import AbbreviationTranslator
+
+
+class AbbreviationTranslatorDictionary(AbbreviationTranslator):
+    """Translates abbreviations in text using a provided dictionary."""
+
+    def __init__(self, pdf_dictionary: dict[str, str]):
+        self.pdf_dictionary = pdf_dictionary
+
+    def translate_abbreviation(self, text: str) -> str:
+        """Translate abbreviations in the text using the provided dictionary."""
+        translated_text = text
+
+        for abbreviation, full_form in self.pdf_dictionary.items():
+            patterns = [
+                (r"\b" + re.escape(abbreviation) + r"s\b", full_form + "s"),
```

Comment:
Here you are checking for short form first and then the long form? Isn't it usually the other way around, at least classic abbreviations which are not chemicals?

Reply:
Did you double check these regex? I'm not sure if this is really what you want. AFAIK we're looking for patterns `very long chemical (VLC)`  and/or `VLC (very long chemical)`. Or is there something else that should be included?